### ISSUE TITLE
 eCo2 and bVOCe is always 0

### ISSUE THREAD

**Issue by Plastefuchs84** [alexh-name/bsec_bme680_linux#19]
Hi, 

since the last updates of 2018 and 2019 he shows me no Air Quality value.
The version BSEC_1.4.6.0_Generic_Release_20180425 Unfortunately, the other version does not work for me. Compiling has worked for all versions.

Maybe they have an idea. Thanks for the great support

pi@raspberrypi:/var/homebridge $ ./bsec_bme680 
bsec_iaq.state empty
2019-10-01 18:54:15,[IAQ (0)]: 25.00,[T degC]: 19.32,[H %rH]: 83.17,[P hPa]: 988.72,[G Ohms]: 344287,[S]: 0,[eCO2 ppm]: 0.00,[bVOCe ppm]: 0.00
2019-10-01 18:54:18,[IAQ (0)]: 25.00,[T degC]: 19.56,[H %rH]: 82.13,[P hPa]: 988.64,[G Ohms]: 458132,[S]: 0,[eCO2 ppm]: 0.00,[bVOCe ppm]: 0.00
2019-10-01 18:54:22,[IAQ (0)]: 25.00,[T degC]: 19.63,[H %rH]: 81.80,[P hPa]: 988.63,[G Ohms]: 483970,[S]: 0,[eCO2 ppm]: 0.00,[bVOCe ppm]: 0.00
2019-10-01 18:54:24,[IAQ (0)]: 25.00,[T degC]: 19.34,[H %rH]: 82.90,[P hPa]: 988.70,[G Ohms]: 460337,[S]: 0,[eCO2 ppm]: 0.00,[bVOCe ppm]: 0.00

**Comment by Plastefuchs84** [comment]
Update: Air Quality works. Sorry :-)))

2019-10-01 19:14:54,[IAQ (1)]: 99.99,[T degC]: 19.97,[H %rH]: 79.45,[P hPa]: 988.64,[G Ohms]: 549533,[S]: 0,[eCO2 ppm]: 0.00,[bVOCe ppm]: 0.00
2019-10-01 19:14:57,[IAQ (1)]: 118.83,[T degC]: 19.69,[H %rH]: 80.59,[P hPa]: 988.66,[G Ohms]: 507069,[S]: 0,[eCO2 ppm]: 0.00,[bVOCe ppm]: 0.00
2019-10-01 19:15:00,[IAQ (1)]: 119.40,[T degC]: 19.96,[H %rH]: 79.43,[P hPa]: 988.68,[G Ohms]: 524143,[S]: 0,[eCO2 ppm]: 0.00,[bVOCe ppm]: 0.00



Unfortunately, the homebridge does not yet speak with the new values
https://github.com/trho/homebridge-bme680

Oct 01 18:40:05 raspberrypi homebridge[932]: [2019-10-1 6:40:05 PM] [Sensor] bsec_iaq.state empty
Oct 01 18:40:07 raspberrypi homebridge[932]: [2019-10-1 6:40:07 PM] [Sensor] unable to parse iaq result: SyntaxError: Unexpected number in JSON at position 4
Oct 01 18:40:10 raspberrypi homebridge[932]: [2019-10-1 6:40:10 PM] [Sensor] unable to parse iaq result: SyntaxError: Unexpected number in JSON at position 4
Oct 01 18:40:13 raspberrypi homebridge[932]: [2019-10-1 6:40:13 PM] [Sensor] unable to parse iaq result: SyntaxError: Unexpected number in JSON at position 4

**Comment by alexh-name** [comment]
Hi,

I have no connection to the repo https://github.com/trho/homebridge-bme680.

Regarding the estimate gas ppm values: This is odd. What version off BSEC are you using? Make sure it's the latest (1.4.7.4).

**Comment by Plastefuchs84** [comment]
Yes i used BSEC 1.4.7.4.

**Comment by Plastefuchs84** [comment]
the result after 10min

2019-10-22 17:48:20,[IAQ (1)]: 25.00,[T degC]: 17.61,[H %rH]: 88.53,[P hPa]: 1011.27,[G Ohms]: 298427,[S]: 0,[eCO2 ppm]: 0.00,[bVOCe ppm]: 0.00
2019-10-22 17:48:23,[IAQ (1)]: 25.00,[T degC]: 17.62,[H %rH]: 88.47,[P hPa]: 1011.27,[G Ohms]: 298157,[S]: 0,[eCO2 ppm]: 0.00,[bVOCe ppm]: 0.00

**Comment by cagriy** [comment]
Testing with BSEC 1.4.7.4 on RPI Zero W.
```
2019-10-28 23:36:32,[IAQ (0)]: 25.00,[T degC]: 20.57,[H %rH]: 54.12,[P hPa]: 1018.81,[G Ohms]: 101709,[S]: 0,[eCO2 ppm]: 500.000000000000000,[bVOCe ppm]: 0.4999999403953552246093750
2019-10-28 23:36:35,[IAQ (0)]: 25.00,[T degC]: 20.57,[H %rH]: 54.11,[P hPa]: 1018.79,[G Ohms]: 101709,[S]: 0,[eCO2 ppm]: 500.000000000000000,[bVOCe ppm]: 0.4999999403953552246093750
2019-10-28 23:36:38,[IAQ (0)]: 25.00,[T degC]: 20.57,[H %rH]: 54.09,[P hPa]: 1018.79,[G Ohms]: 101402,[S]: 0,[eCO2 ppm]: 500.000000000000000,[bVOCe ppm]: 0.4999999403953552246093750```
Basically similar issue, IAQ stuck at (0):25 well after Gas value stabilises. Am I missing something ?

@Plastefuchs84 
What environment these readings were taken ? 80 %rH, and 500K Gas reading ? My Gas reading never goes above 120K.

**Comment by alexh-name** [comment]
Something is off. The values for eCO2/bVOCe should be 500/0.4999 even before reaching BSEC calibration status 1. Can you please post your make.config? Or maybe even your whole project directory?

**Comment by Plastefuchs84** [comment]
pi@raspberrypi:~/bsec_bme680_linux $ ls -l
total 688
-rwxr-xr-x 1 root root 674252 Sep 27 14:11 bsec_bme680
-rw-r--r-- 1 root root   9828 Sep 27 13:16 bsec_bme680.c
-rw-r--r-- 1 pi   pi      458 Sep 27 14:11 bsec_iaq.config
-rw-r--r-- 1 pi   pi        0 Sep 27 13:06 bsec_iaq.state
-rwxrwxrwx 1 root root    439 Sep 27 13:04 make.config
-rwxrwxrwx 1 root root    999 Sep 27 14:07 make.sh
drwxr-xr-x 2 root root   4096 Sep 27 13:37 patches
pi@raspberrypi:~/bsec_bme680_linux $ 

This is the code from make.config:

BSEC_DIR='/usr/src/BSEC_1.4.7.4_Generic_Release'

VERSION='normal_version'

ARCH="${VERSION}/bin/RaspberryPI/PiZero_ArmV6-32bits"

# Other architectures can be found in BSEC_DIR/algo/${VERSION}/bin/.

CONFIG='generic_33v_3s_4d'

# Other configs are:
# generic_18v_300s_28d
# generic_18v_300s_4d
# generic_18v_3s_28d
# generic_18v_3s_4d
# generic_33v_300s_28d
# generic_33v_300s_4d
# generic_33v_3s_28d
# generic_33v_3s_4d

CONFIG_DIR='.'

pi@raspberrypi:/usr/src $ ls -l
total 12
drwxr-xr-x 8 root root 4096 Sep 27 12:41 BSEC_1.4.7.4_Generic_Release
drwxr-xr-x 3 root root 4096 Sep 29 09:14 bsec_bme680_linux
drwxr-xr-x 3 root root 4096 Jul 10 02:14 sense-hat
pi@raspberrypi:/usr/src $

**Comment by Plastefuchs84** [comment]
when compiling also appears an error

pi@raspberrypi:/var/homebridge $ ./make.sh
Patching...
./make.sh: 24: ./make.sh: cannot open patches/eCO2+bVOCe.diff: No such file
Already applied.
Compiling...
Compiled.
Copied config.


UPDATE:

pi@raspberrypi:/var/homebridge $ ./make.sh
Patching...
2 out of 2 hunks FAILED
Already applied.
Compiling...
Compiled.
Copied config.

pi@raspberrypi:/var/homebridge $ ls -l  
total 700
drwxrwxrwx 2 homebridge homebridge   4096 Sep 26 23:51 accessories
-rwxr-xr-x 1 pi         pi         674252 Nov  1 20:29 bsec_bme680
-rw-r--r-- 1 root       root         9828 Oct 29 08:04 bsec_bme680.c
-rwxrwxrwx 1 root       root          458 Nov  1 20:29 bsec_iaq.config
-rw-r--r-- 1 pi         pi              0 Oct  1 18:54 bsec_iaq.state
-rwxrwxrwx 1 homebridge homebridge    489 Oct 26 15:03 config.json
-rwxr-xr-x 1 root       root          469 Sep 30 17:27 make.config
-rwxr-xr-x 1 root       root         1234 Sep 30 17:36 make.sh
drwxr-xr-x 2 pi         pi           4096 Nov  1 20:16 patches
drwxrwxrwx 2 homebridge homebridge   4096 Oct 26 14:58 persist


pi@raspberrypi:/var/homebridge $ ./bsec_bme680 
bsec_iaq.state empty
2019-11-01 20:30:38,[IAQ (0)]: 25.00,[T degC]: 15.58,[H %rH]: 78.97,[P hPa]: 992.68,[G Ohms]: 147968,[S]: 0,[eCO2 ppm]: 0.00,[bVOCe ppm]: 0.00

**Comment by Plastefuchs84** [comment]
hi, I think we can close the question. I've gone back to the old BSEC version for my apple homekit. Thanks anyway

**Comment by TBR-BRD** [comment]
I have the same Problem:
2020-12-08 21:28:18,[IAQ (1)]: 26.55,[T degC]: 20.73,[H %rH]: 61.14,[P hPa]: 1007.06,[G Ohms]: 194726,[S]: 0,[eCO2 ppm]: 0.000000000000000,[bVOCe ppm]: 0.0000000000000000000000000

**Comment by alexh-name** [comment]
Ironically, after I thought that patching bsec_integration.c was not longer necessary in newer versions and just removing the patch, it is needed again. Fixed in 6953631ba3dcd08e8108c27737e31452d6abe768.
